File: backend/credenciales-api/src/main/java/mx/edu/itse/credenciales/entity/Credencial.java
package mx.edu.itse.credenciales.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "credenciales")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Credencial {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String folio;

    @Column(columnDefinition = "TEXT")
    private String codigoQR;

    private String codigoBarras;

    @Column(length = 500)
    private String qrPath;

    @Builder.Default
    private String estado = "ACTIVA";

    @Builder.Default
    private LocalDateTime fechaGeneracion = LocalDateTime.now();

    @OneToOne
    @JoinColumn(name = "alumno_id")
    private Alumno alumno;
}

File: fronted/credenciales-web/src/app/core/models/credencial.ts
export interface Credencial {

  id: number;

  folio: string;

  codigoQR: string;

  codigoBarras: string;

  qrPath: string;

  estado: string;

  fechaGeneracion: string;

  fechaVencimiento?: string;

  alumno: {

    id: number;

    nombreCompleto: string;

    matricula: string;

    semestre: number;

    carrera: {

      id: number;

      nombre: string;

    };

    fotografia: {

      id: number;

      ruta: string;

    };

  };

}